newsroom/topics/topics_async.py

```diff
@@ -0,0 +1,252 @@
+
+        async for topic in topics:
+            updates = dict(
+                subscribers=[s for s in topic.subscribers if s["user_id"] != user_object_id],
+            )
+
+            if topic.user == user_object_id:
+                topic.user = None
+
+            self.update(topic.id, updates)
+
+        # remove user as a topic creator for the rest
+        user_topics = await self.search(lookup={"user": user["_id"]})
+        async for topic in user_topics:
+            await self.update(topic.id, {"user": None})
+
+
+async def get_user_topics(user_id: Union[ObjectId, str, None]) -> List[Topic]:
+    if not user_id:
+        return []
+    user = await UsersService().find_by_id(user_id)
+    data = await TopicService().find(
+        {
+            "$or": [
+                {"user": user.id},
+                {"$and": [{"company": user.company}, {"is_global": True}]},
+            ]
+        }
+    )
+    return await data.to_list_raw()
+
+
+async def get_topics_with_subscribers(topic_type: Optional[str] = None) -> List[Topic]:
+    lookup: Dict[str, Any] = (
+        {"subscribers": {"$exists": True, "$ne": []}}
+        if topic_type is None
+        else {
+            "$and": [
+                {"subscribers": {"$exists": True, "$ne": []}},
+                {"topic_type": topic_type},
+            ]
+        }
+    )
+
+    mongo_cursor = await TopicService().search(lookup=lookup)
+
+    return await mongo_cursor.to_list_raw()
+
+
+async def get_user_id_to_topic_for_subscribers(
+    notification_type: Optional[str] = None,
+) -> Dict[ObjectId, Dict[ObjectId, Topic]]:
+    user_topic_map: Dict[ObjectId, Dict[ObjectId, Topic]] = {}
+    for topic in await get_topics_with_subscribers():
+        for subscriber in topic.get("subscribers") or []:
+            if notification_type is not None and subscriber.get("notification_type") != notification_type:
+                continue
+            user_topic_map.setdefault(subscriber["user_id"], {})
+            user_topic_map[subscriber["user_id"]][topic["_id"]] = topic
+
+    return user_topic_map
+
+
+async def get_agenda_notification_topics_for_query_by_id(item, users):
+    """
+    Returns active topics for a given agenda item
+    :param item: agenda item
+    :param users: active users dict
+    :return: list of topics
+    """
+    lookup = {
+        "$and": [
+            {"subscribers": {"$exists": True, "$ne": []}},
+            {"topic_type": "agenda"},
+            {"query": item["_id"]},
+        ]
+    }
+
+    mongo_cursor = await TopicService().search(lookup=lookup)
+    topics = await mongo_cursor.to_list_raw()
+
+    # filter out the topics those belong to inactive users
+    return [t for t in topics if users.get(str(t["user"]))]
+
+
+async def auto_enable_user_emails(
+    updates: Union[Topic, Dict[str, Any]],
+    original: Union[TopicResourceModel, Dict[str, Any]],
+    user: Optional[Union[User, Dict[str, Any]]],  # Allow user to be None
```

Comment:
```suggestion
    updates: Topic | dict[str, Any],
    original: TopicResourceModel | dict[str, Any],
    user: Optional[User | dict[str, Any]],  # Allow user to be None
```

I briefly talked with @MarkLark86 about the types we should use in a PR review. He mentioned that he's sticking to built-in types as much as possible, then resort back into `typings` and other libs if they'd be needed. It makes sense to me. If you agree too, perhaps we start moving in the direction. Thoughts?

Reply:
makes sense to me